sdk/web-pubsub/web-pubsub-client/src/webPubSubClientCredential.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { GetClientAccessUrlOptions } from "./models";
+
+/**
+ * Type to be used to get clientAccessUri
+ */
+export type ClientAccessUriProvider = (options?: GetClientAccessUrlOptions) => Promise<string>;
+
+/**
+ * The WebPubSubClient credential
+ */
+export interface WebPubSubClientCredential {
```

Comment:
If this is a novel type of credential I wonder if we should add something to `core-auth` 

/cc @schaabs